src/gwc/src/test/java/org/geoserver/gwc/layer/CatalogConfigurationTest.java

```diff
@@ -364,7 +364,7 @@ private void testModifyLayer(GeoServerTileLayer orig) {
         doThrow(new IllegalArgumentException("failedSave")).when(tileLayerCatalog).save(
                 eq(forceState2));
 
-        verify(mockMediator, never()).layerRemoved(anyString());
```

Comment:
If the event gets fired without the tiles being deleted, that's a bug in a component other than the one being tested so I think this is good enough for testing the Configuration.